From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Lawrence <paullawrence@google.com>
Date: Wed, 5 Feb 2020 08:50:12 -0800
Subject: ANDROID: Incremental fs: Fix crash on failed lookup

Don't call dput on error code

Change-Id: Ie63645c9ed67fa231829917ae8ca154e049b4921
Signed-off-by: Paul Lawrence <paullawrence@google.com>
Bug: 133435829
(cherry picked from commit 334164ca0f18ea89a922b90020f5e3840a928503)
---
 fs/incfs/vfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/incfs/vfs.c b/fs/incfs/vfs.c
--- a/fs/incfs/vfs.c
+++ b/fs/incfs/vfs.c
@@ -1481,6 +1481,7 @@ static struct dentry *dir_lookup(struct inode *dir_inode, struct dentry *dentry,
 		err = IS_ERR(backing_dentry)
 			? PTR_ERR(backing_dentry)
 			: -EFAULT;
+		backing_dentry = NULL;
 		goto out;
 	} else {
 		struct inode *inode = NULL;
